doc/source/io.rst

```diff
@@ -440,6 +440,41 @@ individual columns:
     Specifying ``dtype`` with ``engine`` other than 'c' raises a
     ``ValueError``.
 
+Specifying Categorical dtype
+''''''''''''''''''''''''''''
+
+.. versionadded:: 0.19.0
+
+`Categorical` columns can be parsed directly by specifying `dtype='category'`
+
+.. ipython :: python
+
+   data = 'col1,col2,col3\na,b,1\na,b,2\nc,d,3'
+
+   pd.read_csv(StringIO(data))
+   pd.read_csv(StringIO(data)).dtypes
+   pd.read_csv(StringIO(data), dtype='category').dtypes
+
+Individual columns can be parsed as a `Categorical` using a dict specification
+
+.. ipython :: python
+
+   pd.read_csv(StringIO(data), dtype={'col1': 'category'}).dtypes
+
+.. note::
+
+   The resulting categories will always be parsed as string (object dtype).
+   Numeric categories can be converted using the :func:`pd.to_numeric` function.
+
+   .. ipython :: python
+
+      df = pd.read_csv(StringIO(data), dtype='category')
+      df.dtypes
+      df['col3']
+      df['col3'].cat.categories = pd.to_numeric(df['col3'].cat.categories)
```

Comment:
this is not idiomatic, we don't assign directly to ``.categories``, instead use ``.set_categories``

```
In [17]: df = pd.DataFrame({'A' : pd.Categorical(['1','2','1'])})

In [18]: df.A
Out[18]:
0    1
1    2
2    1
Name: A, dtype: category
Categories (2, object): [1, 2]

In [19]: df['A'] = df['A'].cat.set_categories(pd.to_numeric(df['A'].cat.categories), rename=True)

In [20]: df.A
Out[20]:
0    1
1    2
2    1
Name: A, dtype: category
Categories (2, int64): [1, 2]
```

I *could* see adding a ``.apply_categories`` accessor, so this would become:

``df['A'] = df['A'].cat.apply_categories(pd.to_numeric)`` which could do the apply-set-with-rename idiom